airflow/models/mappedoperator.py

```diff
@@ -245,16 +296,13 @@ def _validate_argument_count(self) -> None:
         """
         if isinstance(self.operator_class, str):
             return  # No need to validate deserialized operator.
-        operator = self._create_unmapped_operator(
-            mapped_kwargs={k: unittest.mock.MagicMock(name=k) for k in self.mapped_kwargs},
-            partial_kwargs=self.partial_kwargs,
-            real=False,
-        )
-        if operator.task_group:
-            operator.task_group._remove(operator)
-        dag = operator.get_dag()
+        mocked_mapped_kwargs = create_mocked_kwargs(self.mapped_kwargs)
+        op = self._create_unmapped_operator(mapped_kwargs=mocked_mapped_kwargs, real=False)
```

Comment:
I do wonder if we'll need an escape hatch for this.

I guess we can add that when someone finds a specific case.

Reply:
Escape hatch for if an operator cannot correctly run `__init__` with mocks?

I think eventually we may want to create a separate API for validation without actually creating the operator, perhaps a classmethod called `validate` that can be called here.

Reply:
Yeah, I was thinking an escape hatch for when an Operator ctor does something that causes it to fail when given a Mock.

A proper class-level `validate(**kwargs)` sounds like a better idea.